GliaWidgets/Public/Glia/Glia.swift

```diff
@@ -254,6 +254,10 @@ public class Glia {
 private extension Glia {
     func startObservingInteractorEvents() {
         interactor?.addObserver(self) { [weak self] event in
+            /// Since engagement is accepted before current engagement is established, this case needs to be checked before any other so that visitor code alert can be dismissed properly and visitor code embedded view can trigger callback
+            if case .engagementRequestAccepted = event {
```

Comment:
Don't know what others think but shouldn't it be that rather `self?.callVisualizer` should get the event and decide itself if it can process it, rather than `Glia` determining if call visualizer needs to process the event?

Reply:
That event is exclusive for call visualizer, so Glia must determine it. It is possible however to have an event listener specifically for this event in call visualizer, but since all events would be published to both listeners, I combined them

Reply:
Yes, Glia listens for the events, that is fine. What I meant was something like

```
self?.callVisualizer.handleEventOrSomeSimilarName(event)
```

and then inside callVisualizer

```
fund handleEvent(event: Event) {
    if case .engagementRequestAccepted = event {
        handleEngagementRequestAccepted()
    }
}
```

that way the only entity that knows about this is callVisualizer, which is the only one that should be interested about the fact that we need to do some handling of this event. No knowledge then is spilled into `Glia.swift`.

Reply:
Discussed privately with Rasmus and doing it like this would unfortunately make it inconsistent with the rest of the code so it can just stay as it is. It creates coupling of this code with call visualizer but it is what it is.

Reply:
It's reasonable, but can be a bit confusing in future, because `CallVisualizer` has its own subscription on interactor events and handling these events.